reports/hier/usr/lib/python3/dist-packages/tests/test_reports.py

```diff
@@ -186,6 +192,591 @@ def create_previous_day_table(table_name="sessions", days=1):
         earliest_date = new_date
         days = days - 1
 
+Sql_inject_getDataForReportEntry_parameters = ["ReportEntry", "startDate", "endDate", "extraSelects", "extraConditions", "fromtype", "limit"]
+
+Sql_field_condition_injects = overrides.get(
+    "Sql_field_condition_injects", {
+    "column": [{
+        "javaClass": "com.untangle.app.reports.SqlCondition",
+        "autoFormatValue": False, 
+        # "column": "bypassed is true; {inject};",
+        "column": "bypassed is true {inject}",
+        "operator": "is",
+        "value": "true",
+        "table": None
+    }],
+    # The op field is handled in the SQLCondition object
+    "value": [{
+        "javaClass": "com.untangle.app.reports.SqlCondition",
+        "autoFormatValue": False, 
+        "column": "bypassed",
+        "operator": "is",
+        # "value": "true group by c_client_addr; {inject}; ",
+        "value": "true group by c_client_addr {inject}",
+        "table": None
+    }]
+})
+
+Sql_field_injects = overrides.get(
+    "Sql_field_injects", {
+        "injections":{
+            "semi-colon": "; drop table cmd_exec; create table cmd_exec(cmd_output text); copy cmd_exec from program 'touch {inject_filename}'; ",
+            "dash-comments": "-- \\\\! touch {inject_filename} -- ",
+            "slash-comments": "/* test */ \\\\! touch {inject_filename} /* test */ ",
+            "unmatched-quote": "' ",
```

Comment:
Not sure I follow this one...would "select * from t where c1='text' AND c2=123;" be considered an injection.

Reply:
It's true, we're explicitly limiting semi colons for certain parts of the SQL statement we're constructing.  But most of the WHERE clauses are being "properly" driven through PreparedStatements in Java which handle this problem.